packages/components/src/buttons/src/Button.module.css

```diff
@@ -207,46 +239,122 @@
     --border: var(--hop-Button-upsell-border-disabled);
 }
 
-.hop-Button--upsell[data-hovered] {
+.hop-Button--upsell[data-hovered]:not([data-disabled], [data-loading]) {
     --background-color: var(--hop-Button-upsell-background-color-hover);
     --color: var(--hop-Button-upsell-color-hover);
     --border: var(--hop-Button-upsell-border-hover);
 }
 
-.hop-Button--upsell[data-pressed] {
+.hop-Button--upsell[data-pressed]:not([data-loading]) {
     --background-color: var(--hop-Button-upsell-background-color-pressed);
     --color: var(--hop-Button-upsell-color-pressed);
     --border: var(--hop-Button-upsell-border-pressed);
 }
 
-.hop-Button--negative {
-    --background-color: var(--hop-Button-negative-background-color);
-    --color: var(--hop-Button-negative-color);
-    --border: var(--hop-Button-negative-border);
+.hop-Button--danger {
+    --background-color: var(--hop-Button-danger-background-color);
+    --color: var(--hop-Button-danger-color);
+    --border: var(--hop-Button-danger-border);
+    --spinner-color: var(--hop-Button-danger-spinner-color);
+}
+
+.hop-Button--danger[data-disabled]:not([data-loading]) {
+    --background-color: var(--hop-Button-danger-background-color-disabled);
+    --color: var(--hop-Button-danger-color-disabled);
+    --border: var(--hop-Button-danger-border-disabled);
+}
+
+.hop-Button--danger[data-hovered]:not([data-disabled], [data-loading]) {
```

Comment:
if i'm not mistaken , [data-loading] would apply to all elements with a [data-loading] even outside of our control, this seems dangerous.

Reply:
this selector only targets hop-Button--danger with the attribute data-loading, not it's children

Reply:
I missed the :not() all good